client/src/App.jsx

```diff
@@ -68,7 +118,86 @@ if (gameData && gameData.results) {
 const handleLoginModalVisible = () => {
   setLoginModalVisible(true);
   console.log('login modal is visible', loginModalVisible);
-}
+};
+
+// Function to toggle on newUserModalVisible to make it visible while also turning off loginModalVisible 
+const handleNewUserModalVisible = () => {
+  setNewUserModalVisible(true);
+  setLoginModalVisible(false);
+  console.log('New user modal is visible', newUserModalVisible);
+};
+
+// Function to handle game details modal visibility based on selected game id
+const handleGameDetailsModalVisible = (gameId) => {
+  setSelectedGameId(gameId);
+  setGameDetailsModalVisible(true);
+  console.log('Game details modal is visible', gameDetailsModalVisible);
+  console.log('game details id:', selectedGameId)
+};
+
+// Fetch game details based on game_id
+useEffect(() => {
+  const fetchGameDetails = async () => {
+    if(selectedGameId) {
+      console.log('Fetching details for game id:', selectedGameId);
+      try {
+        const response = await axios.get(`http://localhost:8080/game/${selectedGameId}`);
+        console.log('Game details fetched for:', response.data);
+        setGameDetails(response.data);
+      } catch (error) {
+        console.error('Error fetching game details:', error);
+      }
+    }
+  };
+  fetchGameDetails();
+}, [selectedGameId, newReviewSubmitted]);
+
+console.log('game details from the app', gameDetails);
+
+// Fetch game reviews based on selected game id
+useEffect(() => {
+  const fetchGameReviews = async () => {
+    if(selectedGameId) {
+      console.log('Fetching game reviews for game id;', selectedGameId);
+      try {
+        const response = await axios.get(`http://localhost:8080/game-reviews/${selectedGameId}`);
+        console.log('Game reviews fetched for:', response.data);
+        setGameReviews(response.data || '');
+      } catch (error) {
+        console.error('Error fetching game reviews:', error);
+      }
+    }
+  };
+  fetchGameReviews();
+}, [selectedGameId, newReviewSubmitted]);
+
+console.log('game reviews from the app', gameReviews);
+
+useEffect(() => {
+  console.log('TESTER game reviews from the app:', gameReviews);
+}, [gameReviews]);
+
+// Fetch all user data based on logged-in user id
+useEffect(() => {
+  const fetchAllUserData = async () => {
+    // if(isLoggedIn && loggedInUser.user_id) {
+      if(isLoggedIn, loggedInUser) {
+      console.log('Fetching all logged in user data', loggedInUser.user_id);
+      try {
+        const response = await axios.get(`http://localhost:8080/user-info/${loggedInUser.user_id}`);
+        console.log('User data fetched for:', response.data);
+        setFullLoggedInUserData(response.data);
+      } catch (error) {
+        console.error('Error fetching user data:', error);
+      }
+    }
+  };
+  fetchAllUserData();
+// }, [isLoggedIn, loggedInUser.user_id]);
+}, [isLoggedIn, loggedInUser]);
```

Comment:
Hmm I'll think about that! I had made isLoggedIn first to get the navBar showing buttons correctly based on if a user was logged in, before setting up the actual user info capture. So that may be something I can implement now. Thanks!